### ISSUE TITLE
Duplication bug with filters

### ISSUE THREAD

**Issue by GreninjaSan** [desht/ModularRouters#146]
### Minecraft Version
1.16.5
### Forge Version
36.2.0
### Mod Version
modular-routers-1.16.5-7.5.1-69.jar 
### Describe your problem, including steps to reproduce it
Using insert and Extract, you can duplicate any item in a filter. (See Picture)
### Any other comments?
![image](https://user-images.githubusercontent.com/26509719/130797052-acb2bc36-d3ad-4ed4-95f0-cbbb8a77499d.png)

**Comment by desht** [comment]
Which mod adds those insert and extract buttons?

**Comment by desht** [comment]
Never mind, I figured out it's Quark.

In any case, this is on them, not me.  I've looked at their code, and they just copy items out of the slots / item handler, without checking to see if the slots or underlying item handler actually permit item extraction.  See https://github.com/VazkiiMods/Quark/blob/master/src/main/java/vazkii/quark/base/handler/InventoryTransferHandler.java#L177-L186 for what I mean.

They should either check `Slot#mayPickup()` and/or use `IItemHandler.extractItem()`, not just blindly read the item out of the inventory.
